Marlin/src/sd/SdBaseFile.cpp

```diff
@@ -1506,44 +1512,53 @@ int8_t SdBaseFile::readDir(dir_t *dir, char *longFilename) {
     // Post-process normal file or subdirectory longname, if any
     if (DIR_IS_FILE_OR_SUBDIR(dir)) {
       #if ENABLED(UTF_FILENAME_SUPPORT)
```

Comment:
Although this is the original behavior, I guess UCS-2 to UTF-8 conversion should be done always, since VFat always stores long names as UCS-2 (UTF-16) in LFN sequences, while we always handle 8bit string names.